submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/SubmissionAsyncTest.kt

```diff
@@ -0,0 +1,72 @@
+package ac.uk.ebi.biostd.itest.test.submission.submit
+
+import ac.uk.ebi.biostd.client.integration.commons.SubmissionFormat.TSV
+import ac.uk.ebi.biostd.client.integration.web.BioWebClient
+import ac.uk.ebi.biostd.common.config.PersistenceConfig
+import ac.uk.ebi.biostd.itest.common.SecurityTestService
+import ac.uk.ebi.biostd.itest.entities.SuperUser
+import ac.uk.ebi.biostd.itest.itest.getWebClient
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionPersistenceQueryService
+import ebi.ac.uk.dsl.section
+import ebi.ac.uk.dsl.submission
+import ebi.ac.uk.dsl.tsv.line
+import ebi.ac.uk.dsl.tsv.tsv
+import ebi.ac.uk.extended.mapping.to.ToSubmissionMapper
+import ebi.ac.uk.model.extensions.title
+import org.assertj.core.api.Assertions.assertThat
+import org.awaitility.Awaitility.await
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.web.server.LocalServerPort
+import org.springframework.context.annotation.Import
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import java.time.Duration.ofSeconds
+
+@Import(PersistenceConfig::class)
+@ExtendWith(SpringExtension::class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class SubmissionAsyncTest(
+    @Autowired val securityTestService: SecurityTestService,
+    @Autowired val submissionRepository: SubmissionPersistenceQueryService,
+    @Autowired val toSubmissionMapper: ToSubmissionMapper,
+    @LocalServerPort val serverPort: Int,
+) {
+    private lateinit var webClient: BioWebClient
+
+    @BeforeAll
+    fun init() {
+        securityTestService.ensureUserRegistration(SuperUser)
+        webClient = getWebClient(serverPort, SuperUser)
+    }
+
+    @Test
+    fun `simple submit async`() {
+        val submission = tsv {
+            line("Submission", "SimpleAsync1")
+            line("Title", "Async Submission")
+            line()
+
+            line("RootSection")
+            line()
+        }.toString()
+
+        webClient.submitAsync(submission, TSV)
+
+        await()
+            .atMost(ofSeconds(10))
+            .until {
+                submissionRepository.existByAccNo("SimpleAsync1")
```

Comment:
I think the right query should be `submissionRepository.findByAccNo("SimpleAsync1") != null` since `existByAccNo ` doesn't require for the submission to be active so it could hide any weird behavior